xarray/backends/pseudonetcdf_.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+
+import numpy as np
+
+from .. import Variable
+from ..core.pycompat import OrderedDict
+from ..core.utils import (FrozenOrderedDict, Frozen)
+from ..core import indexing
+
+from .common import AbstractDataStore, DataStorePickleMixin, BackendArray
+
+
+class PncArrayWrapper(BackendArray):
+
+    def __init__(self, variable_name, datastore):
+        self.datastore = datastore
+        self.variable_name = variable_name
+        array = self.get_array()
+        self.shape = array.shape
+        self.dtype = np.dtype(array.dtype)
+
+    def get_array(self):
+        self.datastore.assert_open()
+        return self.datastore.ds.variables[self.variable_name]
+
+    def __getitem__(self, key):
+        key, np_inds = indexing.decompose_indexer(
+            key, self.shape, indexing.IndexingSupport.OUTER_1VECTOR)
+
+        with self.datastore.ensure_open(autoclose=True):
+            array = self.get_array()[key.tuple]  # index backend array
+
+        if len(np_inds.tuple) > 0:
+            # index the loaded np.ndarray
+            array = indexing.NumpyIndexingAdapter(array)[np_inds]
+        return array
+
+
+_genericncf = ('Dataset', 'netcdf', 'ncf', 'nc')
+
+
+class _notnetcdf:
+    def __eq__(self, lhs):
+        return lhs not in _genericncf
+
+
+class PseudoNetCDFDataStore(AbstractDataStore, DataStorePickleMixin):
+    """Store for accessing datasets via PseudoNetCDF
+    """
+    @classmethod
+    def open(cls, filename, format=None, writer=None,
+             autoclose=False, **format_kwds):
+        from PseudoNetCDF._getreader import getreader, getreaderdict
+        readerdict = getreaderdict()
+        _genreaders = tuple([readerdict[rn] for rn in _genericncf
+                             if rn in readerdict])
+        reader = getreader(filename, format=format, **format_kwds)
+        if isinstance(reader, _genreaders):
+            raise ValueError(('In xarray, PseudoNetCDF should not be used ' +
```

Comment:
I think we can avoid this check/error message if we default to `decode_cf=False` for PNC in `open_dataset()`.

Reply:
see previous comment.